src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java

```diff
@@ -378,7 +378,7 @@ public JsonToken nextToken() throws IOException
                 boolean returnEnd = _headContext.isStartHandled();
                 f = _headContext.getFilter();
                 if ((f != null) && (f != TokenFilter.INCLUDE_ALL)) {
-                    f.filterFinishArray();
+                    f.filterFinishObject();
```

Comment:
This handles both `ID_END_ARRAY` and `ID_END_OBJECT` cases so it probably needs to call different method (check `t.id()` again or duplicate)?